fe/fe-core/src/main/java/org/apache/doris/datasource/paimon/PaimonHMSExternalCatalog.java

```diff
@@ -0,0 +1,165 @@
+    }
+
+    @Override
+    protected void initLocalObjectsImpl() {
+        paimonCatalogType = PAIMON_HMS;
+        String metastoreUris = catalogProperty.getOrDefault(HMSProperties.HIVE_METASTORE_URIS, "");
+        String warehouse = catalogProperty.getOrDefault(PaimonProperties.PAIMON_WAREHOUSE, "");
+        Options options = new Options();
+        options.set(PaimonProperties.WAREHOUSE, warehouse);
+        // Currently, only supports hive
+        options.set(PaimonProperties.METASTORE, PaimonProperties.METASTORE_HIVE);
+        options.set(PaimonProperties.URI, metastoreUris);
+        CatalogContext context = CatalogContext.create(options, getConfiguration());
+        try {
+            catalog = create(context);
+        } catch (IOException e) {
+            e.printStackTrace();
+        }
+    }
+
+    public Catalog create(CatalogContext context) throws IOException {
+        Path warehousePath = new Path(context.options().get(CatalogOptions.WAREHOUSE));
+        FileIO fileIO;
+        fileIO = FileIO.get(warehousePath, context);
+        String uri = context.options().get(CatalogOptions.URI);
+        String hiveConfDir = context.options().get(HiveCatalogOptions.HIVE_CONF_DIR);
+        String hadoopConfDir = context.options().get(HiveCatalogOptions.HADOOP_CONF_DIR);
+        HiveConf hiveConf = createHiveConf(hiveConfDir, hadoopConfDir);
+
+        // always using user-set parameters overwrite hive-site.xml parameters
+        context.options().toMap().forEach(hiveConf::set);
+        hiveConf.set(HiveConf.ConfVars.METASTOREURIS.varname, uri);
+        // set the warehouse location to the hiveConf
+        hiveConf.set(HiveConf.ConfVars.METASTOREWAREHOUSE.varname, context.options().get(CatalogOptions.WAREHOUSE));
+
+        String clientClassName = context.options().get(METASTORE_CLIENT_CLASS);
+
+        return new HiveCatalog(fileIO, hiveConf, clientClassName, context.options().toMap());
+    }
+
+    public static HiveConf createHiveConf(@Nullable String hiveConfDir, @Nullable String hadoopConfDir) {
+        Configuration hadoopConf = null;
+        if (!StringUtils.isNullOrWhitespaceOnly(hadoopConfDir)) {
+            hadoopConf = HiveCatalog.getHadoopConfiguration(hadoopConfDir);
+            if (hadoopConf == null) {
+                String possiableUsedConfFiles = "core-site.xml | hdfs-site.xml | yarn-site.xml | mapred-site.xml";
+                throw new RuntimeException("Failed to load the hadoop conf from specified path:" + hadoopConfDir,
+                        new FileNotFoundException("Please check the path none of the conf files ("
+                            + possiableUsedConfFiles + ") exist in the folder."));
+            }
+        }
+
+        if (hadoopConf == null) {
+            hadoopConf = new Configuration();
+        }
+
+        if (hiveConfDir == null) {
+            return new HiveConf(hadoopConf, HiveConf.class);
+        } else {
+            HiveConf.setHiveSiteLocation((URL) null);
```

Comment:
What is these code for? Reading hive-site.xml?

Reply:
Reading hive-site.xml